### PR TITLE
ICMSLST-882 Add goods to Iron and Steel application

### PR DESCRIPTION
**Pull request by kevinetienne**
And update textiles widgets to display commodity group with `get_usage_records`.

### CODE REVIEW

File: web/domains/case/_import/ironsteel/models.py [hunk 1]
```diff
@@ -27,4 +29,48 @@ class IronSteelApplication(ImportApplication):
...
+    )
+
+    goods_description = models.CharField(
+        max_length=100,
+        null=True,
+        verbose_name="Goods Description",
+        help_text="Please describe the goods in no more than five (5) words.",
+    )
+
+    quantity = models.DecimalField(
```
Comment (MattHolmes123) [inline web/domains/case/_import/ironsteel/models.py:69]: We've added quantity like this in a few places (noticed you've said the step to be one in the form)  
```quantity = models.PositiveIntegerField(```

Do we need Decimals or will an integer do for this application type?

File: web/domains/case/_import/ironsteel/widgets.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+from django_select2 import forms as s2forms
+
+from web.domains.case._import.models import ImportApplicationType
+from web.domains.commodity.models import Commodity, CommodityGroup
+from web.domains.country.models import Country
+from web.utils.commodity import get_usage_commodities, get_usage_records
+
+
+class IronSteelCommodityGroupSelect(s2forms.ModelSelect2Widget):
+    queryset = CommodityGroup.objects.filter(commodity_type__type_code="IRON_STEEL")
```
Comment (MattHolmes123) [inline web/domains/case/_import/ironsteel/widgets.py:14]: Does this work when running `reset-migrations.sh` or `reset-db`?

I think I had issues filtering stuff as a model attribute:  
`web/domains/case/_import/derogations/widgets.py`

I can't remember what my exact issue was :/

**Review by oskusalerma — APPROVED** [review]

File: web/static/web/js/pages/iron-steel-goods.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const importer = document.querySelector('#id_origin_country');
+  const categoryDescription = document.querySelector("#category_commodity_group_description");
+
+  const categorySelect2 = $("#id_category_commodity_group").djangoSelect2();
+  const commoditySelect2 = $("#id_commodity").djangoSelect2();
+
+   importer.addEventListener("change", (e) => {
+    categorySelect2.empty();
+    commoditySelect2.empty();
+    categoryDescription.innerHTML = "";
```
Comment (MattHolmes123) [inline web/static/web/js/pages/iron-steel-goods.js:11]: textContent instead of innerHTML

File: web/static/web/js/pages/textiles-edit.js [hunk 4]
```diff
@@ -10,6 +11,7 @@ window.addEventListener('load', (event) => {
   countryOfOrigin.addEventListener("change", (e) => {
     categorySelect.empty();
     commoditySelect.empty();
+    categoryDescription.innerHTML = "";
```
Comment (MattHolmes123) [inline web/static/web/js/pages/textiles-edit.js:14]: textContent instead of innerHTML

**Review by MattHolmes123 — CHANGES_REQUESTED** [review]

File: web/static/web/js/pages/iron-steel-goods.js (hunk 3 above)
Comment (kevinetienne) [reply web/static/web/js/pages/iron-steel-goods.js:11]: Haha! I knew we weren't anymore in the 90's

File: web/domains/case/_import/ironsteel/widgets.py (hunk 2 above)
Comment (kevinetienne) [reply web/domains/case/_import/ironsteel/widgets.py:14]: I've just tried it and it seems fine, leaving it for now if you don't mind.

File: web/static/web/js/pages/iron-steel-goods.js (hunk 3 above)
Comment (kevinetienne) [reply web/static/web/js/pages/iron-steel-goods.js]: Updated in https://github.com/uktrade/icms/pull/519/commits/bb4487b0f1810414858f1cf67e7268ae81d22808

**Review by MattHolmes123 — APPROVED** [review]

**Pull request merged**